regex.go

```diff
@@ -45,24 +44,40 @@ type Regexp struct {
 	errorBuf       *C.char
 	matchData      *MatchData
 	namedGroupInfo NamedGroupInfo
+	mutex          *sync.Mutex
 }
 
 // NewRegexp creates and initializes a new Regexp with the given pattern and option.
-func NewRegexp(pattern string, option int) (re *Regexp, err error) {
+func NewRegexp(pattern string, option int) (*Regexp, error) {
```

Comment:
Just my 2 cents - maybe we can add an extra constructor `NewRegexpWithMutex` where we can pass mutex object.
So people who don't need thread safety (just just single threaded fast regexp) can still use old `NewRegexp` where mutex by default is nil.
WDYT?

Reply:
Ok, nvm. I see it's done opposite `NewRegexp` is thread safe but new one is not.